File: 3069-distribute-elements-into-two-arrays-i/3069-distribute-elements-into-two-arrays-i.java
class Solution {
    public int[] resultArray(int[] nums) {
        int n = nums.length;
      
        // Initialize two arrays to hold the split elements
        int[] firstArray = new int[n];
        int[] secondArray = new int[n];
      
        // Place first element in firstArray and second element in secondArray
        firstArray[0] = nums[0];
        secondArray[0] = nums[1];
      
        // Track the last index of valid elements in each array
        int lastIndexFirst = 0;
        int lastIndexSecond = 0;
      
        // Process remaining elements starting from index 2
        for (int currentIndex = 2; currentIndex < n; currentIndex++) {
            // Compare the last elements of both arrays
            if (firstArray[lastIndexFirst] > secondArray[lastIndexSecond]) {
                // Add current element to firstArray
                lastIndexFirst++;
                firstArray[lastIndexFirst] = nums[currentIndex];
            } else {
                // Add current element to secondArray (when less than or equal)
                lastIndexSecond++;
                secondArray[lastIndexSecond] = nums[currentIndex];
            }
        }
      
        // Merge secondArray into firstArray after all firstArray elements
        for (int index = 0; index <= lastIndexSecond; index++) {
            lastIndexFirst++;
            firstArray[lastIndexFirst] = secondArray[index];
        }
      
        // Return the merged result
        return firstArray;
    }
}

File: 3069-distribute-elements-into-two-arrays-i/3069-distribute-elements-into-two-arrays-i.js
function resultArray(nums) {
    // 첫 번째 배열
    const firstArray = [nums[0]];

    // 두 번째 배열
    const secondArray = [nums[1]];

    // 세 번째 원소부터 처리
    for (const currentElement of nums.slice(2)) {
        // 각 배열의 마지막 원소
        const lastElementFirst =
            firstArray[firstArray.length - 1];

        const lastElementSecond =
            secondArray[secondArray.length - 1];

        // 마지막 원소가 더 큰 배열에 추가
        if (lastElementFirst > lastElementSecond) {
            firstArray.push(currentElement);
        } else {
            secondArray.push(currentElement);
        }
    }

    // 두 배열을 합쳐서 반환
    return firstArray.concat(secondArray);
}